.github/workflows/tetragon-rthook-pr.yaml

```diff
@@ -0,0 +1,83 @@
+name: tetragon-rthooks test
+on:
+  pull_request:
+    paths:
+      - 'contrib/tetragon-rthooks/**'
+  push:
+    branches:
+      - main
+    paths:
+      - 'contrib/tetragon-rthooks/**'
+
+jobs:
+  build:
+    name: Build tetragon-rthooks
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    concurrency:
+      group: ${{ github.ref }}-rthooks-build
+      cancel-in-progress: true
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+      with:
+        path: go/src/github.com/cilium/tetragon/
+
+    - name: Install Go
+      uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
+      with:
+        # renovate: datasource=golang-version depName=go
+        go-version: '1.22.5'
+
+    - name: Build
+      env:
+        GOPATH: /home/runner/work/tetragon/tetragon/go
+      run: |
+        cd go/src/github.com/cilium/tetragon/contrib/tetragon-rthooks
+        make
+
+    - name: tar build
+      run: |
+        cd go/src/github.com/cilium/tetragon/contrib
+        tar cz -f /tmp/tetragon-rthooks.tar ./tetragon-rthooks
+
+    - name: upload build
+      uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+      with:
+         name: tetragon-rthooks-build
+         path: /tmp/tetragon-rthooks.tar
+         retention-days: 1
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+        fail-fast: false
+        matrix:
+           cri:
+            - "containerd"
+            - "crio"
+    concurrency:
+      group: ${{ github.ref }}-rthooks-test-${{ matrix.cri }}
+      cancel-in-progress: true
+    steps:
+    - name: start minikube
+      uses: medyagh/setup-minikube@latest
+      with:
+        driver: docker
+        container-runtime: ${{ matrix.cri }}
+
+    - name: download build data
+      uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
+      with:
+         name: tetragon-rthooks-build
+
+    - name: extract build data
+      run: |
+        tar xf tetragon-rthooks.tar
+
+    - name: run test
+      run: |
+        cd tetragon-rthooks
+        ./scripts/minikube-install-hook.sh
+        ./scripts/minikube-test-hook.sh
```

Comment:
@cilium/github-sec :wave:

Is executing scripts from the repo in a `pull_request` target OK?

Based on offline discussions, I believe it should be OK but I might be missing something and, in any case, would be nice to verify.

Reply:
`pull_request` is the safer option ([docs](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request)) yep. 

Using `pull_request_target` is generally discouraged particularly due to the warning described in the docs: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target 

> Warning: For workflows that are triggered by the pull_request_target event, the GITHUB_TOKEN is granted read/write repository permission unless the permissions key is specified and the workflow can access secrets, even when it is triggered from a fork. Although the workflow runs in the context of the base of the pull request, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event. Additionally, any caches share the same scope as the base branch. To help prevent cache poisoning, you should not save the cache if there is a possibility that the cache contents were altered. For more information, see "[Keeping your GitHub Actions and workflows secure: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests)" on the GitHub Security Lab website.